lib/vanagon/project.rb

```diff
@@ -692,5 +718,36 @@ def load_upstream_settings(upstream_project_name, upstream_git_url, upstream_git
         upstream_project.cleanup
       end
     end
+
+    # Load the settings hash for the current project/platform combination from a
+    # yaml file as produced by `publish_yaml_settings`. file:// and http:// URIs
+    # are accepted. If the URI uses http://, a sha1 URI is also required.
+    #
+    # @param settings_uri [String] A URI to a yaml settings file
+    # @param settings_sha1_uri [String] A URI to a sha1sum file for the yaml settings file
+    # @raise [Vanagon::Error] when the settings file can't be found
+    def load_yaml_settings(settings_uri, settings_sha1_uri = nil) # rubocop:disable Metrics/AbcSize
+      source_type = Vanagon::Component::Source.determine_source_type(settings_uri)
+      if settings_uri =~ /^file/ && source_type == :unknown
```

Comment:
I think it would be better to have this be:
```
if source_type == :unknown
```

instead. You can add an additional note to the exception message if the settings_uri is a local source indicating that the file doesn't exist, so something like:

```
if source_type == :unknown
  message = if settings_ur =~ /^file/ "Tried to load YAML settings from settings uri but file doesn't exist!" else "Unknown source type #{settings_uri}"
  raise Vanagon::Error, message
end
```

Reply:
@ekinanp I've updated the logic as you suggest